modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -109,4 +114,66 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers with Transa
         expect.eql(Left(MalformedTransaction), _)
       )
   }
+
+  test("returns the accountIdentifiers for negative operations") { res =>
+    implicit val (sp, k) = res
+
+    val cs = ConstructionService.make[IO]()
+
+    val gen = for {
+      address <- addressGen
+    } yield address
+
+    forall(gen) {
+      case address =>
+        val result = for {
+          operation <- Some(
+            Operation(
+              OperationIdentifier(OperationIndex(1L)),
+              None,
+              Transfer,
+              None,
+              Some(AccountIdentifier(address, None)),
+              Some(Amount(AmountValue(-1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))))
+            )
+          )
+          accountIdentifiers <- cs.getAccountIdentifiers(operation :: Nil)
+          extractedAddress <- Some(accountIdentifiers.map(id => id.address).head)
+        } yield expect.same(extractedAddress, address)
+
+        result.getOrElse(failure("The addresses are not the same."))
+    }
```

Comment:
1. there is no need to wrap everything into `Some` in the for comprehension. getAccountIdentifiers is a pure function, so it can be simply assigned to `val` or `def`
2. please don't use `expect.same` even if you see such examples in the codebase. we should use `expect.eql` which does not import implicit universal equal (and we are going to change all `expect.same` usages into `expect.eql` in the existing code)
3. in 99% cases you don't need to return `failure` on your own. you wrote an expectation that addresses should be the same, so use this expectation. it will return success or failure automatically

```scala
forall(addressGen) {
  case address =>
    val operation = Operation(
      OperationIdentifier(OperationIndex(1L)),
      None,
      Transfer,
      None,
      AccountIdentifier(address, None).some,
      Amount(AmountValue(-1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))).some
    )
    
    val result = cs.getAccountIdentifiers(List(operation)).map(_.address).head
    
    expect.eql(address, result)
}
```

Reply:
The `.getOrElse(failure("The addresses are not the same."))` is there because the type of `result` is `Option[Expectation]`. To unwrap an `Option`, I only know of `.get` and `.getOrElse` and `.getOrElse` is "less bad". The rest has been cleaned up.

Reply:
I talked with @kpudlik and basically instead of unwrapping result, you can lift an expected address to Option

Reply:
```
    forall(addressGen) {
      case address =>
        val operation = Operation(
          OperationIdentifier(OperationIndex(1L)),
          None,
          Transfer,
          None,
          Some(AccountIdentifier(address, None)),
          Some(Amount(AmountValue(-1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))))
        )

        val expectedAddress: Option[Address] = cs.getAccountIdentifiers(operation :: Nil)
          .map(_.head.address)

        expect.eql(expectedAddress, Some(address)) // You can't compare `Address` with `Option[Address]` but you can lift `Address` to `Option` (wrap by effect so wrap by `Option`) and then you will compare `Option[Address]` with `Option[Address]`
    }
```

Reply:
@cngo-github I see that you didn't change the rest of the things, especially you still use `expect.same`.

```scala
  test("returns the accountIdentifiers for negative operations") { res =>
    implicit val (sp, k) = res

    val cs = ConstructionService.make[IO]()

    forall(addressGen) { address =>
      val operation = Operation(
        OperationIdentifier(OperationIndex(1L)),
        none,
        Transfer,
        none,
        AccountIdentifier(address, none).some,
        Amount(AmountValue(-1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))).some
      )

      val result = cs
        .getAccountIdentifiers(List(operation))
        .map(_.head.address)

      expect.eql(address.some, result)
    }
  }

```

Reply:
I will need guidance on this.

This

```
cs.getAccountIdentifiers(operation :: Nil)
          .map(_.head.address).liftTo[Option]
```

gives me

```
OptionOps.LiftToPartiallyApplied[Option, Address]
```

and I don’t know where to go from there to make it an `Expectation`.

Reply:
please take a look at @kpudlik comment. there is an annotation in the snippet

Reply:
Changed.